### DISCUSSION TITLE
Filename of Photo that is attached to a waypoint

### DISCUSSION THREAD

**Discussion by Maggi68** [osmandapp/OsmAnd#14789] [Ideas]
Hello,
I hope doesnot overlooked a similar discussion....
it would be nice if a taken picture (photo) that would be attached to a waypoint the filename of the picture called like the waypoint at the same way, like it is implamented in Locusmaps. Or what would be also good, if it could be possible to customise the structure of names from taken and attached pictures before the the pictures would be taken, at a settings diaolg, like filename is "waypointname"_Date(yyyy.mm.dd)_Time(hh:mm:ss)". 
Thanks

**Comment by mnalis** [comment]
> it would be nice if a taken picture (photo) that would be attached to a waypoint the filename of the picture called like the waypoint at the same way

And how is it called right now? For me, in OsmAnd~ 4.3.12, that is exactly what happens. I take picture with audio/video plugin, the picture is called e.g. `0IrMdvqo--.1.jpg` and `.gpx` file contains:

```
  <wpt lat="45.4799652" lon="16.3805723">
    <time>2023-04-11T15:35:39Z</time>
    <name>0IrMdvqo--.1.jpg</name>
    <link href="0IrMdvqo--.1.jpg" />
    <extensions />
  </wpt>
```

are you experiencing something different?

> Or what would be also good, if it could be possible to customise the structure of names from taken and attached pictures

Just to note that the "strange" name of picture is actually their geo coordinates encoded in the name using [OSM shortlink / QuadTiles format](https://wiki.openstreetmap.org/wiki/Shortlink), so from name of the picture alone you know exactly where it was taken (e.g. if you system camera does not add geotag your `.jpg` pictures in EXIF headers), so it is actually very useful for your photos and videos and audio files from A/V plugin to be called exactly like they are called.

e.g. if the name of the file is `0IrMdvqo--.1.jpg` and you go to https://osm.org/go/0IrMdvqo-- it will position you at exact coordinates where that picture was taken.

**Comment by Maggi68** [comment]
Thanks for the answer.
But it is not the same I mean in my opinion.
For example: 
I import 60 Waypoints as gpx file.
One of the waypoint is named "Waldsee Argenthal"
![Screenshot_20230422-144543](https://user-images.githubusercontent.com/63550052/233788028-ea617cd2-9a11-4eae-b6dd-e14a572c94a9.png)

So I click at the waypoint and open the menu to take a picture
![Screenshot_20230422-144600](https://user-images.githubusercontent.com/63550052/233788039-fb13dc19-8667-4f15-8fc5-491e0b10182d.png)

When I go to the list of pictures, the picture will be shown with the date and time
![Screenshot_20230422-144827](https://user-images.githubusercontent.com/63550052/233788052-46dfdb04-5128-4b2f-b667-6ded8631d045.png)

If I open the picture at galerie the Filename is called: 0DcoeaWq--.1.jpg
![Screenshot_20230422-144837](https://user-images.githubusercontent.com/63550052/233788064-d2388917-3b2f-4255-9436-e8d1a111cfec.png)

Waht I mean is, the resultng filename should be the same as the waypoint name eg. "Waldsee Argenthal...xxx /date and time-"counternumber").jpg". 
Or best solution, but maybe complicate is OSMANd wrote the Information of the waypoint "Title" and "description" to the Fields "Title" and "description" in the jpeg IPTC Metadata [Standard 2022.1] (https://iptc.org/std/photometadata/examples/IPTC-PhotometadataRef-Std2022.1.jpg) fields of the picture from where the picture is attached when it will be saved (Maybe as a copy from original picture or when it will be shared exported). So the pictures can be searched by Software (Paintprograms or Albumssoftware like Geosetter, Xnview....) at the PC, independent where it is copied or stored. 
Maybe with a IPTC Libraries to read and write 
	PERL:Exiftool (https://exiftool.org/)
	Exiftool Java: https://github.com/mjeanroy/exiftool
like fastphototagger App use it.
Regards

**Reply by mnalis** [reply]
Thanks for clarification, I now understand what you would like. I was talking about the waypoint name that gets encoded in currently recording GPX by OsmAnd, when taking a picture. 

I don't know what the OsmAnd devs think about your idea, but I'll try to give some background about potential problems (according to my quite limited knowledge, so take it with a grain of salt).

OsmAnd A/V Notes plugin **does not have any idea on what you clicked**, much less its name. It always names the files (audio, video, and pictures) according to that location-based [OSM shortlink / QuadTiles format](https://wiki.openstreetmap.org/wiki/Shortlink) at the specified location. And that location is just `lat`+`lon` combination, _without_ any other metadata like "name", and those coordinates could have come from variety of sources:

- clicking on waypoint in some previously loaded gpx (like in your exapmle)
- clicking on any favourite
- doing search and clicking on result
- clicking on any "map marker" flag that you placed on the map
- clicking on any displayed POI in POI overlay
- clicking on any POI on the offline map
- at current GPS location (via `start` / `Take o photo/...`) widget
- by long pressing on any (empty or not) place of the screen
- sharing `geo:` location from another app
- ...

some of them might have metadata which might resemble name/description, and some don't. And supporting each of them would require additional code specific to it. Allowing the user to rename each file manually before saving it might be more easily possible, but it would make taking notes more complicated for everybody, and even users who might appreciate it would face a lot of manual work. 

It would be possible to write a script to postprocess and match pictures to specified GPX file and rename them according to waypoint matches there. 

What would be your use case? I.e. why would you want/need such specific names of `.jpg` files, instead of them being displayed on correct place on the map? (because, perhaps there is another solution that might help you)

**Reply by Maggi68** [reply]
Hello,
thanks for your explanation about the possible problems and different ways to attach a picture. So it is a little bit complicater when I thought. I know this function (as I described to use the "name" of the description to an attached picture with the same filename) from Locusmaps. There is only one way to create the picture as attachment file. (click to waypoint and attach a picture to this).
So my idea was to use the IPTC fields not the filename (it is limited im some way) to search and find a picture that was taken at a special waypoint easier at PC.
The principial idea is, I load some wayponts as gpx file to the app, navigate to the waypoints and take pictures at all waypoints. After this I copy the pictures to the PC, and now I need an easy solution to find the pictures for each waypoint again to copy this in Writer or Word to a text of each waypoint. It is possible to create every filename as the Waypoint, but this manual solution takes a lot of time. So my idea is after I import the waypoints to a navigation app, and attach the pictures to this after I visit this, so the files will be "automatic" named as the waypoint.
I hope I could explain what I mean......
Thanks a lot

**Reply by vshcherb** [reply]
You can export GPX file of Audio Video notes where you have points with A/V names obviously we need to go away from these weird names and we plan to do it in future by combining A/V notes with Favorites
